plugin/src/test/scala/io/github/retronym/dependencyanalyzer/TestUtil.scala

```diff
@@ -0,0 +1,118 @@
+import scalaz.concurrent.Task
+
+object TestUtil {
+
+  import scala.language.postfixOps
+
+  def run(code: String, withDirect: Seq[String] = Seq.empty, withIndirect: Map[String, String] = Map.empty): Seq[String] = {
+    val compileOptions = Seq(
+      constructParam("direct-jars", withDirect),
+      constructParam("indirect-jars", withIndirect.keys),
+      constructParam("indirect-targets", withIndirect.values)
+    ).mkString(" ")
+
+    val extraClasspath = withDirect ++ withIndirect.keys
+
+    val reporter: StoreReporter = runCompilation(code, compileOptions, extraClasspath)
+    reporter.infos.collect({ case msg if msg.severity == reporter.ERROR => msg.msg }).toSeq
+  }
+
+  private def runCompilation(code: String, compileOptions: String, extraClasspath: Seq[String]) = {
+    val fullClasspath: String = {
+      val extraClasspathString = extraClasspath.mkString(":")
+      if (toolboxClasspath.isEmpty) extraClasspathString
+      else s"$toolboxClasspath:$extraClasspathString"
+    }
+    val basicOptions =
+      createBasicCompileOptions(fullClasspath, toolboxPluginOptions)
+
+    eval(code, s"$basicOptions $compileOptions")
+  }
+
+  /** Evaluate using global instance instead of toolbox because toolbox seems
+    * to fail to typecheck code that comes from external dependencies. */
+  private def eval(code: String, compileOptions: String = ""): StoreReporter = {
+    // TODO: Optimize and cache global.
+    val options = CommandLineParser.tokenize(compileOptions)
+    val reporter = new StoreReporter()
+    val settings = new Settings(println)
+    val _ = new CompilerCommand(options, settings)
+    settings.outputDirs.setSingleOutput(new VirtualDirectory("(memory)", None))
+    val global = new Global(settings, reporter)
+    val run = new global.Run
+    val toCompile = new BatchSourceFile("<wrapper-init>", code)
+    run.compileSources(List(toCompile))
+    reporter
+  }
+
+  lazy val baseDir = System.getProperty("user.dir")
+
+  lazy val toolboxClasspath: String = {
+    val jar = System.getProperty("scala.library.location")
+    val libPath = Paths.get(baseDir, jar).toAbsolutePath
+    libPath.toString
+  }
+
+  lazy val toolboxPluginOptions: String = {
+    val jar = System.getProperty("plugin.jar.location")
+    val start= jar.indexOf("/plugin")
+    // this substring is needed due to issue: https://github.com/bazelbuild/bazel/issues/2475
+    val jarInRelationToBaseDir = jar.substring(start, jar.length)
+    val pluginPath = Paths.get(baseDir, jarInRelationToBaseDir).toAbsolutePath
+    s"-Xplugin:${pluginPath} -Jdummy=${pluginPath.toFile.lastModified}"
+  }
+
+  private def createBasicCompileOptions(classpath: String, usePluginOptions: String) =
+    s"-classpath $classpath $usePluginOptions"
+
+  private def constructParam(name: String, values: Iterable[String]) = {
+    if (values.isEmpty) ""
+    else s"-P:dependency-analyzer:$name:${values.mkString(":")}"
+  }
+
+  object Coursier {
```

Comment:
@johnynek That is correct, we noted this issue in the description...
This coming week, a couple of Wix developers are working on this feature.
Their first task is to move the tests in this unit test class to the the **E2E** test run by the `test_run` script. this way using _Coursier_ will not be neccessary

As this is is the only issue waiting to be resolved in the coming week, we would appreciate it if you can give your OK to merge this PR, in order for the new developers to work on a new branch out of master